web/todo.py

```
@@ -1,18 +1,16 @@
+def generate_todo_api(user_id: int = Depends(get_current_user)):
+    saved = todo_service.generate_todo(user_id)
+    return {"success": True, "todos": saved}
 
-@router.get("/todo")
-def get_todos():
-    todo_data = TodoData()
-    data = todo_data.get_todos()
-    return {"success": True, "data": data}
\ No newline at end of file
+@router.patch("/todos/{todo_id}")
+def update_todo(todo_id: int, payload: dict = {}, user_id: int = Depends(get_current_user)):
```

Ruff (0.14.7)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)